### PR TITLE
Coriant Network Hardware Page.

### PR DESCRIPTION
**Pull request by xbeaudouin**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

#### Coriant TNMS support

This will add a new page that allow Coriant Hardware to be seen and get the right status.

### CODE REVIEW

**Comment by laf** [comment]
I did mention in the other PR that it wouldn't be good to see mef-this, mef-that poller/discovery modules - this is what I mean. How comes this is separate, what is it actually giving the user?

**Comment by laf** [comment]
Also the sql schema file should be all on one line.

**Comment by xbeaudouin** [comment]
@laf: the SQL is fixed. About MEF, this is constructor idenpendant stuff, totaly outside Coriant. So that's why this is yet another poller/discovery stuff.

**Comment by murrant** [comment]
What exactly does this bring?  Could components be used?

**Comment by laf** [comment]
That doesn't explain what this gives though?

**Comment by xbeaudouin** [comment]
Here some screen shot, that explain better the output.
capture d ecran 2017-03-15 a 16 01 25

Coriant TNMS is somewhat an proxy to several devices, this give us status to each devices and allow us to raize alerts :)

**Comment by murrant** [comment]
Do you think any other device os will benefit from this module?

If not, it might just make sense to put this in the os polling file `includes/polling/os/coriant.inc.php`

**Comment by Rosiak** [comment]
@murrant 
Wireless controllers? Not saying it directly could benefit from this MEC stuff, but the general approach could perhaps be used? List connected AP's, their state etc.

File: html/pages/device.inc.php [hunk 1]
```diff
@@ -173,6 +173,14 @@
                 </li>';
         }
 
+        if (@dbFetchCell("SELECT COUNT(id) FROM tnmsneinfo WHERE device_id = '".$device['device_id']."'") > '0') {
```
Comment (murrant) [inline html/pages/device.inc.php]: Check the os before running a query.  Like this: `if ($device['os'] == 'coriant' &&` here

**Comment by murrant** [comment]
Perhaps, this should either be made a little more generic or not be a module.

**Comment by xbeaudouin** [comment]
I am opened to suggestions. Coriant snmp stuff is really a mess IMHO

**Comment by laf** [comment]
Maybe a module called hardware?

**Comment by xbeaudouin** [comment]
@laf: as you wish, I am opened to all suggestions on your side.

**Comment by laf** [comment]
Let's get a consensus on it first.

**Comment by murrant** [comment]
I feel like just putting this in the os polling module is correct it doesn't seem like it can be shared with other os.

**Comment by xbeaudouin** [comment]
Since Coriant SNMP implementation is mostly working only on Coriant (the snmp this stuff is kinda strange BTW), I don't think we can share anything on other OS / hardware...

**Comment by murrant** [comment]
So you agree, move the module polling code to `includes/polling/os/coriant.inc.php`

**Comment by laf** [comment]
I personally don't as os module should be simply for that purpose - yes I know we have other cruft in now but that will need tidying up as well.

**Comment by murrant** [comment]
It seems like a reasonable place for something like this.  I don't want a poller module for everything.

**Comment by laf** [comment]
Neither do I but this seems like an excessive use of the os poller module.

**Comment by xbeaudouin** [comment]
@laf: the Coriant TNMS SNMP is somewhat brain damage, I can provide in private the documentation of this. Since this PR is mostly the begining of the support.

**Comment by guilhermefigueiredo** [comment]
Hello!

 How to do to test and add a coriant host in my librenms? I already did github-apply 6187 but apparently do not have the option yet .. remembering that we need the context in snmpv3, could they help me?

**Comment by laf** [comment]
I've pushed an update to this to move the module into the os polling instead.

For me to approve this needs the devices table page moving to http://www.jquery-bootgrid.com/. We shouldn't have unpaginaged tables.

I also don't see why we'd have SqlSNMP(), looks like the function wrapper can be removed and we can just call the code.

**Comment by murrant** [comment]
I also, think this should be converted to use "Components":
http://docs.librenms.org/Extensions/Component/

**Comment by xbeaudouin** [comment]
@guilhermefigueiredo: you need to have access to TNMS system and enable SNMP
@laf: the "SqlNMP" function is a basis of the way Coriant store the data... eg : 
capture d ecran 2017-03-14 a 15 00 54
(Yes this is like a database)
@murrant: Will have a look about Components, thanks for the hint.

**Comment by laf** [comment]
@murrant I'm not a huge fan of components, it makes alerting that much more difficult for people to understand and use.

@xbeaudouin I don't see the relevance still. All you are doing is running an snmpwalk and looping through the data to insert/delete/update.

**Comment by xbeaudouin** [comment]
@laf: Started to read the Components, I am not sure this can be used with Coriant SNMP. I have several snmp tables with relational data between them. The fact is I have to add all data into a DB to make join between them to get some more data. I have some documentation from Coriant that really give me an headache. I can show you in private

**Comment by guilhermefigueiredo** [comment]
@xbeaudouin ,

I have access to TNMS and snmp is already configured (version 3), but I need to add the "context" (-n option in snmpwalk) or else I can not do the collection.

root@nms:/opt/librenms# snmpwalk -v3 -a MD5 -l authPriv -u Administrator -A passWorD -x AES -X passWorD X.X.Y.Y
Error in packet.
Reason: noAccess
root@nms:/opt/librenms# 

root@nms:/opt/librenms# snmpwalk -v3 -a MD5 -n tnms -l authPriv -u Administrator -A passWorD -x AES -X passWorD X.X.Y.Y
SNMPv2-MIB::sysDescr.0 = STRING: P42022-P5173-C547-01-50VB hiT 7300 5.40 70.02.00 IC0220101160620 Copyright 2016 Coriant. All rights reserved.                   
SNMPv2-MIB::sysObjectID.0 = OID: SNMPv2-SMI::enterprises.42229.1.1.2863311536
DISMAN-EVENT-MIB::sysUpTimeInstance = Timeticks: (846062685) 97 days, 22:10:26.85
SNMPv2-MIB::sysContact.0 = STRING: eng
SNMPv2-MIB::sysName.0 = STRING: CORIANT-F
SNMPv2-MIB::sysLocation.0 = STRING: F-SITE
SNMPv2-MIB::sysServices.0 = INTEGER: 1
SNMPv2-MIB::sysORLastChange.0 = Timeticks: (0) 0:00:00.00
Error in packet.
Reason: noAccess
Failed object: SNMPv2-MIB::sysORLastChange.0

root@nms:/opt/librenms#

**Comment by xbeaudouin** [comment]
@guilhermefigueiredo: hum... I don't use snmp v3 with them (mostly because our TNMS is behind a private network), did you try with v2c ? 
If it work you can ask them to fix their software ....

**Comment by guilhermefigueiredo** [comment]
@xbeaudouin So I understand hiT 7300 does not work with snmp v2c, only with v3

**Comment by murrant** [comment]
I think the core design for components is good, but it could use some Polish.

**Comment by xbeaudouin** [comment]
@guilhermefigueiredo:  hum... Seems we don't same version of TNMS server. I have TNMS version 16.10.4.24.981 on a Linux server. What is yours ?

**Comment by guilhermefigueiredo** [comment]
@xbeaudouin my version is 16 on a windows 7, but is the TNMS-NCT I think this version is some limited. We are trying to see with the coriant how to release access using snmp v2c.

**Comment by laf** [comment]
Any update on this?

Imho, this needs the tables moving to bootgrid and the code moving out of the function. I know you say it will change later but from everything you said I'm just not sure how practical some of the additional code will be.

**Comment by xbeaudouin** [comment]
@laf: what do you mean by bootgrid ? (I wasn't sure this is for me).

**Comment by laf** [comment]
Basically we want to have paginated tables so that we don't end up with 100's of items on one page. It's not that difficult to do.

See:

`html/pages/peering/ix-list.inc.php`
`html/includes/table/ix-list.inc.php`

If you get stuck I can probably do it for you.

**Comment by xbeaudouin** [comment]
@laf: I started to convert it. Mostly working, but I have issue with formating with nice colors the labels like in the screen shot. Since I am not a guru of javascript... :p

**Comment by xbeaudouin** [comment]
@laf: just finished the conversion into bootgrid :p

**Comment by laf** [comment]
image

**Pull request merged**